WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/usecases/FetchScanHistoryUseCaseTest.kt

```diff
@@ -0,0 +1,106 @@
+import org.wordpress.android.test
+import org.wordpress.android.ui.jetpack.scan.usecases.FetchScanHistoryUseCase.FetchScanHistoryState.Failure
+import org.wordpress.android.ui.jetpack.scan.usecases.FetchScanHistoryUseCase.FetchScanHistoryState.Success
+import org.wordpress.android.util.NetworkUtilsWrapper
+import org.wordpress.android.util.analytics.ScanTracker
+
+@InternalCoroutinesApi
+class FetchScanHistoryUseCaseTest : BaseUnitTest() {
+    private lateinit var fetchScanHistoryUseCase: FetchScanHistoryUseCase
+
+    @Mock private lateinit var networkUtilsWrapper: NetworkUtilsWrapper
+    @Mock private lateinit var scanStore: ScanStore
+    @Mock private lateinit var scanTracker: ScanTracker
+    private val site: SiteModel = SiteModel()
+
+    @Before
+    fun setUp() {
+        fetchScanHistoryUseCase = FetchScanHistoryUseCase(networkUtilsWrapper, scanStore, scanTracker, TEST_DISPATCHER)
+
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(true)
+    }
+
+    @Test
+    fun `Network failure returned, when the device is offline`() = test {
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(false)
+
+        val result = fetchScanHistoryUseCase.fetch(site)
+
+        assertThat(result).isInstanceOf(Failure.NetworkUnavailable::class.java)
+    }
+
+    @Test
+    fun `Network error tracked, when the device is offline`() = test {
+        whenever(networkUtilsWrapper.isNetworkAvailable()).thenReturn(false)
+
+        fetchScanHistoryUseCase.fetch(site)
+
+        verify(scanTracker).trackOnError(ScanTracker.ErrorAction.FETCH_SCAN_HISTORY, ScanTracker.ErrorCause.OFFLINE)
+    }
+
+    @Test
+    fun `Request failure returned, when the request fails`() = test {
+        whenever(scanStore.fetchScanHistory(anyOrNull())).thenReturn(
+                OnScanHistoryFetched(site.siteId, FetchScanHistoryError(GENERIC_ERROR), FETCH_SCAN_HISTORY)
+        )
+
+        val result = fetchScanHistoryUseCase.fetch(site)
+
+        assertThat(result).isInstanceOf(Failure.RemoteRequestFailure::class.java)
+    }
+
+    @Test
+    fun `Request failure tracked, when the request fails`() = test {
+        whenever(scanStore.fetchScanHistory(anyOrNull())).thenReturn(
+                OnScanHistoryFetched(site.siteId, FetchScanHistoryError(GENERIC_ERROR), FETCH_SCAN_HISTORY)
+        )
+
+        fetchScanHistoryUseCase.fetch(site)
+
+        verify(scanTracker).trackOnError(ScanTracker.ErrorAction.FETCH_SCAN_HISTORY, ScanTracker.ErrorCause.REMOTE)
+    }
+
+    @Test
+    fun `Success returned, when the request succeeds`() = test {
+        whenever(scanStore.fetchScanHistory(anyOrNull())).thenReturn(
+                OnScanHistoryFetched(site.siteId, null, FETCH_SCAN_HISTORY)
+        )
+
+        val result = fetchScanHistoryUseCase.fetch(site)
+
+        assertThat(result).isInstanceOf(Success::class.java)
+    }
+
+    @Test
+    fun `Data from db returned, when the request suceeds`() = test {
```

Comment:
1) Minor (🔍): There is a typo in `suceeds`.
2) Suggestion (💡): Consider removing the above `isInstanceOf` test since this assert logic in included within this test as well (`result as Success`). I understand that you wanted to split the test and have them do exactly one thing, but in that case a `Success` has no real value without the list of threat models and as such I would consider them the same (as you did with this test). If the `Success` returned value and the `scanStore` logic was doing two separate things I would indeed have them as separate test, but in this instance they both work for the same single result. Wdyt?

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/13994/commits/aa14dcdffc630331fef6d69b83b4f71ec6600bc3